src/sentry/static/sentry/app/stores/modalStore.tsx

```diff
@@ -25,13 +25,6 @@ const ModalStore = Reflux.createStore({
   },
 
   onCloseModal() {
-    const onClose = this.state?.options?.onClose;
-
-    // Trigger the options.onClose callback
-    if (typeof onClose === 'function') {
```

Comment:
this is already being called in `handleCloseModal` of the `GlobalModal`  when the `onHide` prop is fired. 

https://github.com/getsentry/sentry/blob/19823b5b9235d24add77df2122267ca007619ea6/src/sentry/static/sentry/app/components/globalModal.tsx#L47

but the main reason behind this removal was a problem when switching routes. The problem was that the `onClose` function related exclusively to the current `location.pathname` was being fired on the route new route causing the UI to go back to the previous route.  

Removing this code makes the `onClose` fire only when the `onHide` of the modal is also fired.

Reply:
Could this removal cause the `ModalActions.closeModal()` and `closeModal()` function of the modalActions module misbehave? It looks like this was the only path they had to trigger the onclose hook.

Reply:
Me and @priscilawebdev took a look at this together. I'm not quite sure why I made this change here https://github.com/getsentry/sentry/pull/18245, the PR description makes sense, but based on Priscila's testing it looks like the bootstrap onHide is correctly called when we close the modal no matter what.

I'm not 100% confident that it might not cause some regression, but it seems OK.